interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/ComponentIdProvider.scala

```diff
@@ -1,47 +1,22 @@
-package pl.touk.nussknacker.ui.component
+package pl.touk.nussknacker.engine.definition
 
-import com.typesafe.scalalogging.LazyLogging
-import pl.touk.nussknacker.engine.ProcessingTypeData
 import pl.touk.nussknacker.engine.api.component.ComponentType.ComponentType
 import pl.touk.nussknacker.engine.api.component.{ComponentId, ComponentType}
 import pl.touk.nussknacker.engine.component.ComponentUtil
 import pl.touk.nussknacker.engine.component.ComponentsUiConfigExtractor.ComponentsUiConfig
 import pl.touk.nussknacker.engine.graph.node.{NodeData, WithComponent}
-import pl.touk.nussknacker.restmodel.process.ProcessingType
-import pl.touk.nussknacker.ui.process.ProcessCategoryService
 
 //TODO: It is work around for components duplication across multiple scenario types, until we figure how to do deduplication.
 trait ComponentIdProvider {
-  def createComponentId(processingType: ProcessingType, name: Option[String], componentType: ComponentType): ComponentId
-  def nodeToComponentId(processingType: ProcessingType, node: NodeData): Option[ComponentId]
+  def createComponentId(processingType: String, name: Option[String], componentType: ComponentType): ComponentId
+  def nodeToComponentId(processingType: String, node: NodeData): Option[ComponentId]
 }
 
-object DefaultComponentIdProvider extends LazyLogging {
-
-  def createUnsafe(
-      processingTypeDataMap: Map[ProcessingType, ProcessingTypeData],
-      categoryService: ProcessCategoryService
-  ): ComponentIdProvider = {
-    logger.debug("Creating component id provider")
-
-    val componentObjectsService = new ComponentObjectsService(categoryService)
-    val componentObjectsMap     = processingTypeDataMap.transform(componentObjectsService.prepareWithoutFragments)
-    val componentIdProvider = new DefaultComponentIdProvider(componentObjectsMap.transform {
-      case (_, componentsObjects) => componentsObjects.config
-    })
-
-    ComponentsValidator.checkUnsafe(componentObjectsMap, componentIdProvider)
-
-    componentIdProvider
-  }
-
-}
-
-class DefaultComponentIdProvider(configs: Map[ProcessingType, ComponentsUiConfig]) extends ComponentIdProvider {
+class DefaultComponentIdProvider(configs: Map[String, ComponentsUiConfig]) extends ComponentIdProvider {
 
   override def createComponentId(
-      processingType: ProcessingType,
-      name: Option[ProcessingType],
+      processingType: String,
```

Comment:
Why do you replace this alias with String?

Reply:
when moving the the class to a different module, the typedef went out of scope, I didn't choose to duplicate/move it

i'm not a fan of `types` that are just aliases for String, in my opinion it creates more confusion than value
also there are plenty of cases in the code where processingType is just String already

moving the alias definition higher (to common-api for example) generates changes in 50+ files, I can do it in a separate PR

Reply:
I see now, I didn't realize that you moved `ComponentIdProvider` into the interpreter. It's fine, I have the same feeling about type aliases. I would like to replace this type alias with just a value class like we do e.g. with `ProcessName` and similar - we can do this in the separate PR.